xarray/core/combine.py

```diff
@@ -207,7 +212,10 @@ def _dataset_concat(datasets, dim, data_vars, coords, compat, positions):
 
     dim, coord = _calc_concat_dim_coord(dim)
     datasets = [as_dataset(ds) for ds in datasets]
-    datasets = align(*datasets, join='outer', copy=False, exclude=[dim])
+    if not prealigned:
+        datasets = align(*datasets, join='outer', copy=False, exclude=[dim])
+    else:
+        coords = 'minimal'
```

Comment:
It's bad form to unilaterally override an argument with another value -- it's better to raise an error (or maybe a warning).

The only value of `coords` that really breaks here is `'different'`, and even that value could conceivably make sense.